code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -248,57 +356,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	if(isliving(loc))
 		M.IgniteMob()
 	smoketime--
-	if(smoketime < 1)
-		new type_butt(location)
-		processing_objects.Remove(src)
+	if(smoketime <= 0)
+		new type_butt(location) //Spawn the cigarette butt
+		lit = 0 //Actually unlight the cigarette so that the lighting can update correctly
+		update_brightness()
 		if(ismob(loc))
 			M << "<span class='notice'>Your [name] goes out.</span>"
-			M.u_equip(src,0)	//un-equip it so the overlays can update
+			M.u_equip(src, 0)	//Un-equip it so the overlays can update
 			M.update_inv_wear_mask(0)

```

Comment:
You don't need this line, since it's automatically handled in u_equip for humans.